Drop-In/src/main/java/com/braintreepayments/api/DropInActivity.java

```diff
@@ -414,20 +408,53 @@ public void finish() {
     @Override
     protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
         super.onActivityResult(requestCode, resultCode, data);
+        switch (requestCode) {
+            case BraintreeRequestCodes.THREE_D_SECURE:
+                handleThreeDSecureActivityResult(resultCode, data);
+            case BraintreeRequestCodes.GOOGLE_PAY:
+                handleGooglePayActivityResult(resultCode, data);
+            case BraintreeRequestCodes.VENMO:
+                handleVenmoActivityResult(resultCode, data);
+        }
+    }
+
+    private void handleThreeDSecureActivityResult(int resultCode, Intent data) {
         getDropInClient().handleThreeDSecureActivityResult(this, resultCode, data, new DropInResultCallback() {
             @Override
             public void onResult(@Nullable DropInResult dropInResult, @Nullable Exception error) {
-                if (dropInResult != null) {
-                    finishWithDropInResult(dropInResult);
-                } else if (error instanceof UserCanceledException) {
-                    dropInViewModel.setUserCanceledError(error);
-                } else {
-                    onError(error);
-                }
+                onDropInResult(dropInResult, error);
             }
         });
     }
 
+    private void handleGooglePayActivityResult(int resultCode, Intent data) {
+        getDropInClient().handleGooglePayActivityResult(this, resultCode, data, new DropInResultCallback() {
+            @Override
+            public void onResult(@Nullable DropInResult dropInResult, @Nullable Exception error) {
+                onDropInResult(dropInResult, error);
+            }
+        });
+    }
+
+    private void handleVenmoActivityResult(int resultCode, Intent data) {
+        getDropInClient().handleVenmoActivityResult(this, resultCode, data, new DropInResultCallback() {
+            @Override
+            public void onResult(@Nullable DropInResult dropInResult, @Nullable Exception error) {
+                onDropInResult(dropInResult, error);
+            }
+        });
+    }
+
+    private void onDropInResult(DropInResult dropInResult, Exception error) {
```

Comment:
Not sure if this is the best name for this method. Open to other ideas here.

Reply:
I'm not opposed to this name it makes sense coming from a `DropInResultCallback` I feel.

Reply:
Agreed!